exec/java-exec/src/main/java/org/apache/drill/exec/store/http/HttpOAuthConfig.java

```diff
@@ -0,0 +1,173 @@
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@JsonDeserialize(builder = HttpOAuthConfig.HttpOAuthConfigBuilder.class)
+public class HttpOAuthConfig {
+
+  @JsonProperty("baseURL")
+  private final String baseURL;
+
+  @JsonProperty("clientID")
+  private final String clientID;
+
+  @JsonProperty("clientSecret")
+  private final String clientSecret;
+
+  @JsonProperty("callbackURL")
+  private final String callbackURL;
+
+  @JsonProperty("authorizationURL")
+  private final String authorizationURL;
+
+  @JsonProperty("authorizationPath")
+  private final String authorizationPath;
+
+  @JsonProperty("authorizationParams")
+  private final Map<String, String> authorizationParams;
+
+  @JsonProperty("accessTokenPath")
+  private final String accessTokenPath;
+
+  @JsonProperty("generateCSRFToken")
+  private final boolean generateCSRFToken;
+
+  @JsonProperty("scope")
+  private final String scope;
+
+  @JsonProperty("tokens")
+  private final Map<String, String> tokens;
+
+  /**
+   * Clone constructor used for updating tokens
+   * @param that The original oAuth configs
+   * @param tokens The updated tokens
+   */
+  public HttpOAuthConfig(HttpOAuthConfig that, Map<String, String> tokens) {
+    this.baseURL = that.baseURL;
+    this.clientID = that.clientID;
+    this.clientSecret = that.clientSecret;
+    this.callbackURL = that.callbackURL;
+    this.authorizationURL = that.authorizationURL;
+    this.authorizationPath = that.authorizationPath;
+    this.authorizationParams = that.authorizationParams;
+    this.accessTokenPath = that.accessTokenPath;
+    this.generateCSRFToken = that.generateCSRFToken;
+    this.scope = that.scope;
+    this.tokens = tokens;
+  }
+
+  private HttpOAuthConfig(HttpOAuthConfig.HttpOAuthConfigBuilder builder) {
+    this.baseURL = builder.baseURL;
+    this.clientID = builder.clientID;
+    this.clientSecret = builder.clientSecret;
+    this.callbackURL = builder.callbackURL;
+    this.authorizationURL = builder.authorizationURL;
+    this.authorizationPath = builder.authorizationPath;
+    this.authorizationParams = builder.authorizationParams;
+    this.accessTokenPath = builder.accessTokenPath;
+    this.generateCSRFToken = builder.generateCSRFToken;
+    this.scope = builder.scope;
+    this.tokens = builder.tokens;
+  }
+
+  @Override
+  public String toString() {
+    return new PlanStringBuilder(this)
+      .field("baseURL", baseURL)
+      .field("clientID", clientID)
+      .maskedField("clientSecret", clientSecret)
+      .field("callbackURL", callbackURL)
+      .field("authorizationURL", authorizationURL)
+      .field("authorizationParams", authorizationParams)
+      .field("authorizationPath", authorizationPath)
+      .field("accessTokenPath", accessTokenPath)
+      .field("generateCSRFToken", generateCSRFToken)
+      .field("tokens", tokens.keySet())
+      .toString();
+  }
+
+  @JsonPOJOBuilder(withPrefix = "")
+  public static class HttpOAuthConfigBuilder {
+    @Getter
+    @Setter
```

Comment:
Instead of adding annotations for every field, you can add them to the class and they will be applied to every field.

Reply:
Fixed.